src/core/helpers/ObjectHash.ts

```diff
@@ -1,7 +1,90 @@
 import * as crypto from 'crypto-js';
+import { inject, named } from 'inversify';
+import { HashableObjectType } from '../../api/enums/HashableObjectType';
+import { ListingItemFactory } from '../../api/factories/ListingItemFactory';
+import { ImageFactory } from '../../api/factories/ImageFactory';
+import { ItemCategoryService } from '../../api/services/ItemCategoryService';
+import { MarketService } from '../../api/services/MarketService';
+import { Types, Core, Targets } from '../../constants';
+import { Logger as LoggerType } from '../../core/Logger';
+import { ItemImageDataCreateRequest } from '../../api/requests/ItemImageDataCreateRequest';
+import { ImageVersions } from '../../core/helpers/ImageVersionEnumType';
+
 
 export class ObjectHash {
-    public static getHash(obj: any): string {
-        return crypto.SHA256(JSON.stringify(obj).split('').sort().toString()).toString();
+
+    public log: LoggerType;
+
+    constructor(
+        @inject(Types.Factory) @named(Targets.Factory.ListingItemFactory) public listingItemFactory: ListingItemFactory,
+        @inject(Types.Service) @named(Targets.Service.ItemCategoryService) public itemCategoryService: ItemCategoryService,
+        @inject(Types.Service) @named(Targets.Service.MarketService) public marketService: MarketService,
+        @inject(Types.Factory) @named(Targets.Factory.ImageFactory) public imageFactory: ImageFactory,
+        @inject(Types.Core) @named(Core.Logger) Logger: typeof LoggerType
+    ) {
+        this.log = new Logger(__filename);
+    }
+
+    public async getHash(obj: any, type: HashableObjectType): Promise<string> {
+        let revisedObj;
+        switch (type) {
+            case HashableObjectType.LISTINGITEM: {
+                const rootCategoryWithRelatedModel: any = await this.itemCategoryService.findRoot();
+                const rootCategory = rootCategoryWithRelatedModel.toJSON();
+                let market;
+                if (obj.Market) {
+                    market = obj.Market;
+                } else {
+                    market = await this.marketService.getDefault();
+                }
+
+                // call factory for getting filter data(without id's just like message)
+                revisedObj = await this.listingItemFactory.getModel(obj, market.id, rootCategory);
+                break;
+            }
+            case HashableObjectType.LISTINGITEMTEMPLATE: {
+                // todo
```

Comment:
function getHash(objectToHash, type)

case HashableObjectType.LISTINGITEM: 
case HashableObjectType.LISTINGITEMTEMPLATE:
    templateOrItem = objectToHash
    hashableObject = factory.getMessage(templateOrItem, templateOrItemCategoryWithRelated)
    delete hashableObject.hash
    break
case HashableObjectType.ITEMIMAGE:
    // create the hash from ORIGINAL imageversion
    imageData = objectToHash.getOriginalImageVersionData()
    hashableObject = {
        protocol: imageData.protocol, 
        encoding: imageData.encoding,
        data: imageData.data,
        id: imageData.dataId
    }
    break
return hash(hashableObject)